tools/shaderc/shaderc.cpp

```diff
@@ -1179,34 +1208,38 @@ namespace bgfx
 				preprocessor.setDefine(glslDefine);
 			}
 		}
-		else if (0 == bx::strCmpI(platform, "asm.js") )
-		{
-			preprocessor.setDefine("BX_PLATFORM_EMSCRIPTEN=1");
-			preprocessor.setDefine(glslDefine);
-		}
-		else if (0 == bx::strCmpI(platform, "ios") )
+		else if (0 == bx::strCmpI(platform, "linux"))
 		{
-			preprocessor.setDefine("BX_PLATFORM_IOS=1");
-			if (profile->lang == ShadingLang::Metal)
+			preprocessor.setDefine("BX_PLATFORM_LINUX=1");
+			if (profile->lang == ShadingLang::SpirV)
 			{
-				preprocessor.setDefine("BGFX_SHADER_LANGUAGE_METAL=1");
+				preprocessor.setDefine("BGFX_SHADER_LANGUAGE_SPIRV=1");
 			}
 			else
 			{
 				preprocessor.setDefine(glslDefine);
 			}
 		}
-		else if (0 == bx::strCmpI(platform, "linux") )
+		else if (0 == bx::strCmpI(platform, "asm.js") )
 		{
-			preprocessor.setDefine("BX_PLATFORM_LINUX=1");
-			if (profile->lang == ShadingLang::SpirV)
-			{
-				preprocessor.setDefine("BGFX_SHADER_LANGUAGE_SPIRV=1");
-			}
```

Comment:
Why reordering?

Reply:
I've reorganized the code for better usability. Now that the iOS and macOS code paths share almost identical configurations, it's highly likely that changes in one will require corresponding adjustments in the other. With this in mind, I felt it was helpful to group these paths together, making it easier to locate related code.

Initially, I considered merging the two code paths, but I hesitated, thinking it might deviate too much from the current platform selection format. However, now that we're discussing it, what are your thoughts on this idea ? Here's the idea in code:

```cpp
else if (0 == bx::strCmpI(platform, "ios") || 0 == bx::strCmpI(platform, "osx") )
{
	if (0 == bx::strCmpI(platform, "ios")
	{
		preprocessor.setDefine("BX_PLATFORM_IOS=1");
	}
	else
	{
		preprocessor.setDefine("BX_PLATFORM_OSX=1");
	}
		
	if (profile->lang != ShadingLang::Metal)
	{
		preprocessor.setDefine(glslDefine);
	}
	char temp[32];
	bx::snprintf(
		temp
		, sizeof(temp)
		, "BGFX_SHADER_LANGUAGE_METAL=%d"
		, (profile->lang == ShadingLang::Metal) ? profile->id : 0
	);
	preprocessor.setDefine(temp);
}
```

On a related note, I wanted to ask: should we continue supporting GLSL on iOS/macOS with `shaderc` ? If not, I'm more than willing to remove the current `if (profile->lang != ShadingLang::Metal)` code path.

Reply:
shaderc should be able to cross-compile for other targets on any platform. This obviously is not true for HLSL, but that's due D3D compiler constraint of being Windows only, not design choice.